Add a floating user info menu

vyneer opened the pull request:
This PR adds a floating user info menu that can be accessed by right-clicking a username in chat. The menu shows all the user's flairs/features (hidden flairs are shown as text), selected message(s) and some action buttons: right now you can mute, ban, check logs of ([just opens RustleSearch](https://rustlesearch.dev/?username=vyneer&channel=Destinygg)), open whispers w/, and ignore a user.

https://user-images.githubusercontent.com/41237021/198858154-118dd750-3e82-498a-be58-95c53e82a3d5.mp4

The mute and ban buttons are only available for admins/moderators, regular users just see the last 3 buttons.

![ksnip_20221030-035831](https://user-images.githubusercontent.com/41237021/198858197-d87e3f75-3042-4e9b-8bae-edc340ab252f.png)

11k commented on assets/chat/js/menus.js:
```diff
@@ -452,10 +453,289 @@ class ChatWhisperUsers extends ChatMenu {
+        return features !== '' ? `<span class="features">${features}</span>` : ''
+    }
+
+    hasModPowers(user) {
```
I think the `ChatUser` class is a better fit for this method.

vyneer replied:
Good call! Moved it :+1:

11k commented:
I spent more time thinking about the action buttons and have some suggestions. Right now, the ban and mute actions are just a slightly more user-friendly way to execute a slash command. Let's instead take the approach that banning or muting via the user menu should be a shortcut to executing the commands.

With this in mind,
1. Rather than a duration field, the mute action submenu should have separate buttons for common mute lengths: 1m, 10m, 1h, and 1d. Simply clicking on one of the buttons immediately mutes the user.
2. Bans should work the same way, with durations of 1d, 7d, 30d, and Perm. You can auto-generate a reason for the ban to make this work, the same way the chat bot does. Additionally, the bans should be IP bans. You can remove support for regular bans.

Separate from the above, the URL used to navigate to a user's logs page should be configurable for better compatibility with Dgg clone websites. You should be able to pass a URL that contains a placeholder, such as `%USERNAME%`, into the `Chat` constructor. When the logs action button is clicked, it should replace the placeholder with a username and navigate to the page. If a logs URL is not provided, you can just not render the button at all.

11k commented:
Actually, forget the bit with configuring a logs URL. Let's just have the logs button close the menu and execute a `/stalk` for the user to keep things simple.

vyneer commented:
Replaced the text input stuff with the duration buttons:

![image](https://user-images.githubusercontent.com/41237021/200023636-1ca0a47d-1d33-4ec4-a2fd-a5f76f13f4b8.png)

They have the same hover highlight effect as the other buttons, think it looks _okay_, but if you have ideas on how to make it prettier I'm all ears!

Changed up the logs button to redirect to /stalk as well. Good thing you asked me to do that, because I found a stupid bug that broke /mute and /ban commands. Shouldn't've removed the ```static``` from the duration calculator function :smile: 

Also made some other minor changes: formatting memes plus started reusing cmdMUTE and cmdBAN from chat.js instead of copy pasting it.
